pennylane/workflow/qnode.py

```diff
@@ -451,19 +464,19 @@ def circuit_unpacking(x):
 
     def __init__(
         self,
-        func,
+        func: Callable,
         device: Union[Device, "qml.devices.Device"],
-        interface="auto",
-        diff_method="best",
-        expansion_strategy=None,
-        max_expansion=None,
-        grad_on_execution="best",
-        cache="auto",
-        cachesize=10000,
-        max_diff=1,
-        device_vjp=False,
-        postselect_mode=None,
-        mcm_method=None,
+        interface: SupportedInterfaceUserInput = "auto",
+        diff_method: Union[TransformDispatcher, SupportedDiffMethods] = "best",
+        expansion_strategy: Literal[None, "device", "gradient"] = None,
+        max_expansion: Optional[int] = None,
+        grad_on_execution: Literal[True, False, "best"] = "best",
+        cache: Union[MutableMapping, Literal["auto", True, False]] = "auto",
+        cachesize: int = 10000,
+        max_diff: int = 1,
+        device_vjp: bool = False,

```

Comment:
```suggestion
        device_vjp: Union[None, bool] = False,
```
`device_vjp=None` means "leave it up to the device to choose".